src/internet_identity/src/anchor_management/registration.rs

```diff
@@ -118,11 +120,30 @@ fn create_captcha<T: RngCore>(rng: T) -> (Base64, String) {
     return (resp, captcha.chars_as_string());
 }
 
+lazy_static! {
+    /// Map of problematic characters that are easily mixed up by humans to the "normalized" replacement.
+    /// I.e. the captcha will never contain any of the characters in the key set and any input provided
+    /// will be mapped to the matching value.
+    /// Note: the captcha library already excludes the characters o, O and 0.
+    static ref CHAR_REPLACEMENTS: HashMap<char, char> = vec![
+        ('C', 'c'),
+        ('l', '1'),
+        ('X', 'x'),
+        ('Y', 'y'),
+        ('Z', 'z'),
+    ].into_iter().collect();
+}
+
+const CAPTCHA_LENGTH: usize = 5;
 #[cfg(not(feature = "dummy_captcha"))]
 fn create_captcha<T: RngCore>(rng: T) -> (Base64, String) {
     let mut captcha = captcha::RngCaptcha::from_rng(rng);
+    let mut chars = captcha.supported_chars();
+    chars.retain(|c| !CHAR_REPLACEMENTS.contains_key(c));
+
     let captcha = captcha
-        .add_chars(5)
+        .set_chars(&chars)
+        .add_chars(CAPTCHA_LENGTH as u32)
```

Comment:
Would it be simpler to normalize the challenge result instead?

Reply:
The challenge result is normalized. But I can also have the captcha generated _with_ ambiguous characters and normalize both, the challenge and the input by the user when checking the result. Is this what you would prefer?

Reply:
Ah my bad, I was confused by `CAPTCHA_LENGTH`. I thought you were adding as many chars to the captcha as there were in `CHAR_REPLACEMENTS` and then only got more confused from there.

If the challenge result is normalized, why also replace `chars`? Is that just for consistency?

Reply:
I think we have 2 options:
1. Normalize (reduce) input chars, such that the captcha does not contain ambiguous chars (i.e. `1lXcx` will never be generated) and also normalize the input by the user.
2. Generate Captcha as before (i.e. `1lXcx`) and only when checking the solution normalize _both_ the actual challenge and the user input. So in this example the user input would be normalize and checked against `11xcx`.

This PR so far implements approach 1.